lib/fd.js

```diff
@@ -0,0 +1,309 @@
+const fs = require('fs')
+const errors = require('./errors')
+const byteStream = require('byte-stream')
+const through = require('through2')
+const pumpify = require('pumpify')
+const messages = require('./messages')
+
+const noop = () => {}
+
+const {
+  O_RDONLY,
+  O_WRONLY,
+  O_RDWR,
+  O_CREAT,
+  O_TRUNC,
+  O_APPEND,
+  O_SYNC,
+  O_EXCL
+} = fs.constants
+const O_ACCMODE = 3
```

Comment:
is this not defined in fs.constants?

Reply:
Got rid of the `fs` dependency (replaced it with `filesystem-constants`, as per offline discussion).